crates/sui-metrics-proxy/src/handlers.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use axum::{
+    body::Body,
+    extract::State,
+    http::{header, Request, StatusCode},
+    response::IntoResponse,
+};
+use std::sync::Arc;
+use tokio::sync::mpsc::Sender;
+use tracing::error;
+
+use crate::channels::NodeMetric;
+
+/// Publish handler which receives metrics from nodes.  Nodes will call us at this endpoint
+/// and we relay them to the upstream tsdb
+///
+/// An mpsc is used within this handler so that we can immediately return an accept to calling nodes.
+/// Downstream processing failures may still result in metrics being dropped.
+pub async fn publish_metrics(
+    State(state): State<Arc<Sender<NodeMetric>>>,
+    request: Request<Body>,
```

Comment:
You should be able to use a different Extractor (String, Json, etc) to do the handling of parsing the body into the format you want. This will simplify the handler logic

Reply:
I intentionally didn't parse to json, that would slow it down and we'll throw away the parsed results.  Getting the body as bytes and relaying it upstream with minimal work down to it was the goal.  Is there a faster way to do this w/o parsing the body (to get the bytes) as i've done here?

Reply:
AFAIK we'll need to parse the payload because we'll also need to insert our own labels to the data prior to sending it further upstream. We probably also want to avoid sending bogus data upwards?

Reply:
I need to see the data that is relayed to me before i know how to change it or what needs to be changed, but it's possible i might need to decode it.  I was hoping that we could not rewrite the entire json blob to add labels, but if it comes to that, then yeah. I'll need to add more tests to simulate sending that prom data and create it synthetically for the test i think.

Reply:
I have the labels being added now. There is still some work to do with integrating it with the chain data, but i'd like to do that in the next PR.